homeassistant/components/device_tracker/huawei_router.py

```diff
@@ -0,0 +1,169 @@
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.util import Throttle
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=30)
+
+_LOGGER = logging.getLogger(__name__)
+
+ARRAY_REGEX = re.compile(r'var UserDevinfo = new Array\((.*),null\);')
+DEVICE_REGEX = re.compile(r'new USERDevice\((.*?)\),')
+DEVICE_ATTR_REGEX = re.compile('"(?P<Domain>.*?)","(?P<IpAddr>.*?)",'
+                               '"(?P<MacAddr>.*?)","(?P<Port>.*?)",'
+                               '"(?P<IpType>.*?)","(?P<DevType>.*?)",'
+                               '"(?P<DevStatus>.*?)","(?P<PortType>.*?)",'
+                               '"(?P<Time>.*?)","(?P<HostName>.*?)",'
+                               '"(?P<IPv4Enabled>.*?)","(?P<IPv6Enabled>.*?)",'
+                               '"(?P<DeviceType>.*?)"')
+LOGIN_COOKIE = dict(Cookie='body:Language:portuguese:id=-1')
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string
+})
+
+
+# pylint: disable=unused-argument
+def get_scanner(hass, config):
+    """Validate the configuration and return a HUAWEI scanner."""
+    scanner = HuaweiDeviceScanner(config[DOMAIN])
+
+    return scanner
+
+
+class Device:
+    """This class represents a network device."""
+
+    def __init__(self, name, ip, mac, state):
+        """Init the network device."""
+        self._name = name
+        self._ip = ip
+        self._mac = mac
+        self._state = state
+
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    def ipaddr(self):
+        """Return the IP address of the device."""
+        return self._ip
+
+    def macaddr(self):
+        """Return the MAC address of the device."""
+        return self._mac
+
+    def state(self):
+        """Return the state of the device (true -> online)."""
+        return self._state
+
+
+class HuaweiDeviceScanner(DeviceScanner):
+    """This class queries a router running HUAWEI firmware."""
+
+    def __init__(self, config):
+        """Initialize the scanner."""
+        self.host = config[CONF_HOST]
+        self.username = config[CONF_USERNAME]
+        self.password = base64.b64encode(bytes(config[CONF_PASSWORD], 'utf-8'))
+
+        self.lock = threading.Lock()
+
+        self.last_results = []
+
+    def scan_devices(self):
+        """Scan for new devices and return a list with found device IDs."""
+        self._update_info()
+        return [client.macaddr() for client in self.last_results]
+
+    def get_device_name(self, device):
+        """Return the name of the given device or None if we don't know."""
+        if not self.last_results:
+            return None
+        for client in self.last_results:
+            if client.macaddr() == device:
+                return client.name()
+        return None
+
+    @Throttle(MIN_TIME_BETWEEN_SCANS)
```

Comment:
Remove this, see commends about threaded. That value is handled by user, But if the value of 30sec is importet you can stay it.